src/coverage-system/renderer.ts

```diff
@@ -143,17 +140,13 @@ export class Renderer {
         if (!section || !section.branches) {
             return;
         }
-        // TODO cleanup this area by using maps, filters, etc
-        section.branches.details.forEach((detail) => {
-            if (detail.taken === 0) {
-                if (detail.line < 0) { return; }
+        section.branches.details.filter((detail) => detail.taken === 0 && detail.line >= 0).forEach((detail) => {
```

Comment:
I like this filter, but we might want to put it on multiple lines to aid in readability (if possible).
```ts
  section.branches.details
  .filter((detail) => detail.taken === 0 && detail.line >= 0)
  .forEach((detail) => {
      const partialRange = new Range(detail.line - 1, 0, detail.line - 1, 0);
      // Evaluates to true if at least one element in range is equal to partialRange
      if (coverageLines.full.some((range) => range.isEqual(partialRange))){
          coverageLines.full = coverageLines.full.filter((range) => !range.isEqual(partialRange));
          coverageLines.partial.push(partialRange);
      }
  });
```